index.js

```diff
@@ -57,7 +60,7 @@ async function subsetFont(
     exports.hb_set_add(inputUnicodes, c.codePointAt(0));
   }
 
-  const subset = exports.hb_subset(face, input);
+  const subset = exports.hb_subset_or_fail(face, input);
```

Comment:
now that it can return zero, feel free to check it and fail gracefully if it returns 0.

Reply:
also maybe on `hb_subset_input_create_or_fail` or any API named '_or_fail` meaning they can return zero, but it is very unlikely.

Reply:
That makes sense! Is there a way to get a better error than "it failed" when that happens?

Reply:
Oh, if the convention here is to return exception as below, that can be done here also, it should be unlikely anyway, still.

Reply:
Something like bb2a0c59ab65ef23ee3ff66711a3a2608eed750b?

Reply:
I don't remember whether `hb_face_reference_blob` can work with a null input, guess for more brevity you can just leave it alone for now and just have it mind :)

Reply:
The `try { throw... } finally { ... }` construct will see to it that we destroy the input, but don't execute the `const result = exports.hb_face_reference_blob(subset);` line?

Reply:
something like that, now I regret bringing it up! a cleanup just like below `if (subsetByteLength === 0) {` would be cool I think :)

Reply:
But I already did implement it like that? Doesn't it look OK as-is? 😼

Reply:
I meant I think before throw these are needed also,

```
    exports.hb_face_destroy(face);
    exports.free(fontBuffer);
```

Reply:
Ah, I get it, fixed in e2deb41